### PR TITLE
Feature/route navigation

### PR DESCRIPTION
**Pull request by Ishadijcks**
Route navigation works now, supports multiple regions.

### CODE REVIEW

File: src/scripts/GameConstants.ts [hunk 1]
```diff
@@ -74,5 +90,59 @@ namespace GameConstants {
...
+            22: [1],
+            23: [22],
+            24: [4],
+            25: [24],
+        }
+    };
+
+    export const routeBadgeRequirements = {
+        0: {
+            3: GameConstants.Badge.Boulder,
```
Comment (dennism1997) [inline src/scripts/GameConstants.ts]: the accesToRoute method does not seem to work when the index is not specified (so for example route 1,2,4 etc)

`Uncaught TypeError: Cannot read property '1' of undefined
    at Function.RouteRequirement.accessToRoute (script.min.js:2900)
    at <anonymous>:1:18`

File: src/scripts/Player.ts [hunk 2]
```diff
@@ -3,22 +3,27 @@
...
+    private static _dungeonTokens: number = 0;
     private static _caughtPokemonList = [];
-    private static _route: number;
-    private static _region: GameConstants.Region;
-    private static _pokeballs: number[] = [3, 2, 4, 1];
+    private static _route: number = 1;
+    private static _routeKills: number[] = Array.apply(null, Array(GameConstants.AMOUNT_OF_ROUTES)).map(Number.prototype.valueOf, 0);
+    private static _routeKillsNeeded: number = 10;
+    private static _region: GameConstants.Region = GameConstants.Region.kanto;
+    private static _gymBadges: GameConstants.Badge[] = [GameConstants.Badge.Boulder];
```
Comment (dennism1997) [inline src/scripts/Player.ts]: Maybe an array of booleans is easier? Then we can use
`if(_gymbadges[0]) do stuff`

Comment (Ishadijcks) [inline src/scripts/Player.ts]: It's easier and faster, but with an array of booleans we can't catch the undefined badge.
`Player.hasBadge(badge)` works correctly now, and can be used.

Comment (dennism1997) [inline src/scripts/Player.ts]: That's right

**Pull request merged**